### PR TITLE
Content management promote

### PR DESCRIPTION
**Pull request by hustodemon**
## What does this PR change?

This PR adds the "promote" functionality to the Content Project, which allows propagating built sources through the Content Environments.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- [doc-susemanager](https://github.com/SUSE/doc-susemanager) PR or issue was created (GitHub automatic link expected below)

- [ ] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Fixes #
Tracks https://github.com/SUSE/spacewalk/issues/6518

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/taskomatic/task/CobblerSyncTask.java [hunk 1]
```diff
@@ -67,6 +67,7 @@ public void execute(JobExecutionContext ctxIn)
                     MethodUtil.getClassFromConfig(CobblerXMLRPCHelper.class.getName());
 
             Double mtime = null;
+            if (1 == 1) return;
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/taskomatic/task/CobblerSyncTask.java]: what is this ?

Comment (hustodemon) [reply java/code/src/com/redhat/rhn/taskomatic/task/CobblerSyncTask.java]: It's a [WIP] pull request.

**Comment by lneves12** [comment]
This was merged into the wrong branch :rofl: I was like: where the hell did the promote code go?

I opened a new PR to merge it to the right branch. https://github.com/uyuni-project/uyuni/pull/761

**Comment by hustodemon** [comment]
Ufff, sorry for that :sweat_smile:  I shouldn't have done it so late in the evening. Thanks for taking care of this.

**Pull request merged**